### PR TITLE
fully implement puppeteer@2.1.1/lib/Connection.js 

### PR DESCRIPTION
**Pull request by Granitosaurus**
There are few spots that need some testing.

### CODE REVIEW

File: pyppeteer/connection.py [hunk 1]
```diff
@@ -21,6 +27,28 @@
 logger_session = logging.getLogger(__name__ + '.CDPSession')
```
Comment (Mattwmaster58) [inline pyppeteer/connection.py]: I wonder what the reasoning behind this was? Doesn't really matter at this point

File: pyppeteer/connection.py [hunk 2]
```diff
@@ -40,14 +68,27 @@ def __init__(
         self._lastId = 0
         self._callbacks: Dict[int, asyncio.Future] = dict()
```
Comment (Mattwmaster58) [inline pyppeteer/connection.py]:
```suggestion
        self._callbacks: Dict[int, asyncio.Future] = {}
```

File: pyppeteer/connection.py [hunk 3]
```diff
@@ -40,14 +68,27 @@ def __init__(
...
         self._delay = delay / 1000
+
         self._loop = loop
         self._sessions: Dict[str, CDPSession] = dict()
```
Comment (Mattwmaster58) [inline pyppeteer/connection.py]:
```suggestion
        self._sessions: Dict[str, CDPSession] = {}
```

File: pyppeteer/connection.py [hunk 4]
```diff
@@ -40,14 +68,27 @@ def __init__(
...
         self.connection: CDPSession
         self._connected = False
         self._ws = websockets.client.connect(
-            self._url, max_size=None, loop=self._loop, ping_interval=None, ping_timeout=None)
+            self._url,
+            max_size=None,
+            loop=self._loop,
+            ping_interval=None,
+            ping_timeout=None,
+        )
```
Comment (Mattwmaster58) [inline pyppeteer/connection.py]: This becomes irrlevant with the work on launcher.py (where this is used)

**Review by Mattwmaster58 — CHANGES_REQUESTED** [review]
Nice work. All type hint of `'CDPSession'` can be replaced with `CDPSession` since you imported it anyway IINM

File: pyppeteer/connection.py (hunk 4 above)
Comment (Granitosaurus) [reply pyppeteer/connection.py]: update to use the new WebsocketTransport class for this.

File: pyppeteer/connection.py (hunk 1 above)
Comment (Granitosaurus) [reply pyppeteer/connection.py]: AFAIK It's for debug logging send/received. It could probably be nicer though.

Comment (Mattwmaster58) [reply pyppeteer/connection.py]: Yeah I'm aware, I was actually just wondering why '.CDPSession' was appended to the name of the module (none of the other modules feature this logging style)

Comment (Granitosaurus) [reply pyppeteer/connection.py]: I left only the connection logger as CDPSession logger is redundant.

**Pull request merged**